data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OffsetManager.java

```diff
@@ -88,12 +89,32 @@ public OffsetManager(
                 logPartitions("revoked", partitions);
             }
         };
+
+        this.partitionAssignedHandler = partitions -> {
+            this.assignedPartitions = partitions;
+
+            return Future.succeededFuture().mapEmpty();
+        };
+
+        this.offsetTrackerCleanupTimerId = vertx.setPeriodic(OFFSET_TRACKER_CLEANUP_INTERVAL_MS, h -> {
+            for (var entry : offsetTrackers.entrySet()) {
+                var tp = entry.getKey();
+                if (!assignedPartitions.stream().anyMatch(o -> o.partition() == tp.partition())) {
```

Comment:
should this compare the topic names as well? (not just the partitions, which are just the integers)

Reply:
oh. Is it possible to have the same partition"Id" on different topics?